### PR TITLE
Add a default converter plugin

### PR DESCRIPTION
**Pull request by mikebauer**
## Description of change
Add a plugin which performs default conversions for ROS2 types to ReSim Types.

## Guide to reproduce test results.
```
full_test () {                                                                                      
   TARGET="$1"                                                                                      
   VALGRIND="valgrind --leak-check=full --show-leak-kinds=all --track-origins=yes --error-exitcode=1"

   bazel test --runs_per_test=100 "${TARGET}" && \                                                  
   bazel test --test_timeout=3600 --run_under="${VALGRIND}" "${TARGET}"       
}
full_test //resim/msg:default_converter_plugin_test
```
## Checklist:
- [x] I have self-reviewed this change.
- [x] I have tested this change.
- [x] This change is covered by tests that are already landed, or in this PR.

### CODE REVIEW

File: resim/msg/default_converter_plugin.cc [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+#include <cstdint>
+#include <cstring>
+#include <functional>
+#include <rclcpp/serialization.hpp>
+#include <rclcpp/serialized_message.hpp>
+#include <string>
+#include <string_view>
```
Comment (knowlsie) [inline resim/msg/default_converter_plugin.cc:13]: Is this used?

File: resim/msg/default_converter_plugin.cc [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+  rclcpp::SerializedMessage message{*ros2_message};
+  serialization.deserialize_message(&message, &deserialized);
+  const auto converted_message = convert_from_ros2(deserialized);
+
+  const std::size_t converted_size = converted_message.ByteSizeLong();
+
+  rcl_serialized_message_t &result = *resim_message;
+  result.allocator = ros2_message->allocator;
+  // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+  result.buffer = reinterpret_cast<uint8_t *>(
```
Comment (knowlsie) [inline resim/msg/default_converter_plugin.cc:65]: I should have commented this on previous PR, but we should make clear that this allocate is allowed and will be deallocated by `ConverterPlugin::convert(...)`

File: resim/msg/default_converter_plugin_test.cc [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+    InOut<std::mt19937> rng) {
+  auto pose = random_element<transforms::proto::SE3>(rng);
+  // We don't use frame IDs when converting to/from ROS2.
+  constexpr int DIMS = 3;
+  pose.mutable_into()->mutable_id()->set_data(
+      transforms::Frame<DIMS>::null_frame().id().to_string());
+  pose.mutable_from()->mutable_id()->set_data(
+      transforms::Frame<DIMS>::null_frame().id().to_string());
+  return pose;
+}
```
Comment (knowlsie) [inline resim/msg/default_converter_plugin_test.cc:144]: "generics save me so much time!" - probably @mikebauer at some point.

**Review by knowlsie — APPROVED** [review]
Looks great!

File: resim/msg/default_converter_plugin.cc (hunk 1 above)
Comment (mikebauer) [reply resim/msg/default_converter_plugin.cc:13]: Nope! I've removed it.

File: resim/msg/default_converter_plugin.cc (hunk 2 above)
Comment (mikebauer) [reply resim/msg/default_converter_plugin.cc:65]: Good call. I've added some comments on this to converter_plugin.hh

File: resim/msg/default_converter_plugin_test.cc (hunk 3 above)
Comment (mikebauer) [reply resim/msg/default_converter_plugin_test.cc:144]: Everyday!

**Pull request merged**